core/schema/modulesource.go

```diff
@@ -145,25 +163,130 @@ func (s *moduleSchema) moduleSource(ctx context.Context, query *core.Query, args
 	return src, nil
 }
 
+// shared logic with originToPath, copy-pasted
+// to avoid dependency on Buildkit
```

Comment:
Not sure I understand this comment - this function uses `gitutil` which already has a dep on buildkit.

I'm happy with the impl though.